### PR TITLE
modified the Introduction and added few more points -- OLM

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
Document an introduction to using OLM concentrating on the following core:

1. Brief high level core benefit that describes how using OLM solves cluster admins and developers needs
2. Introduce the building blocks needed to get started as quickly as possible with OLM
3. Link to Getting Started and Advanced doc links

### CODE REVIEW

**Comment by theishshah** [comment]
Double check and clean up links/references to ensure that they are pointing to the correct paths (the last couple are pointed at the old docs/olm-book)

File: content/en/docs/_index.md [hunk 1]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+Once you've [created your operator's package manifests](packaging-an-operator.md), you will want to ensure that your package is valid and in the correct format. To ensure this, you should take several steps to ensure that your package can be used to install your operator in OLM.
+
+### [Publish Package into catalog](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/make-operator-part-of-catalog.md)
+
+We will publish the package into a catalog, install that catalog onto a Kube cluster, and then install the operator onto that cluster. If all of that succeeds and your operator is behaving as expected, your package is valid.
+
+### [Subscriptions](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Concepts/crds/subscription.md)
+
+Subscriptions are Custom Resources that relate an operator to a CatalogSource. Subscriptions describe which channel of an operator package to subscribe to and whether to perform updates automatically or manually. If set to automatic, the Subscription ensures OLM will manage and upgrade the operator to ensure the latest version is always running in the cluster.
```
Comment (rashmigottipati) [inline content/en/docs/_index.md:78]: Creating a subscription will install an operator as well. Not just limited to upgrades.

Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Yes, I modified the subscription information.

**Review by theishshah — APPROVED** [review]

**Review by rashmigottipati — APPROVED** [review]
/lgtm

File: content/en/docs/_index.md [hunk 2]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### [Publish Package into catalog](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/make-operator-part-of-catalog.md)
+
+We will publish the package into a catalog, install that catalog onto a Kube cluster, and then install the operator onto that cluster. If all of that succeeds and your operator is behaving as expected, your package is valid.
+
+### [Subscriptions](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Concepts/crds/subscription.md)
+
+A Subscription represents an intention to install an operator. Subscriptions are Custom Resources that relate an operator to a CatalogSource. Subscriptions describe which [channel](/docs/glossary/#channel) of an operator package to subscribe to and whether to perform updates automatically or manually. If set to automatic, the Subscription ensures OLM will manage and upgrade the operator to ensure the latest version is always running in the cluster.
```
Comment (jmrodri) [inline content/en/docs/_index.md:78]: THIS is exactly what I'd expect in a "Building blocks of OLM", basically concepts that exist in OLM an what they do.

File: content/en/docs/_index.md [hunk 3]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### [Install Operator](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/install-operator-with-olm.md)
+
+Once you've made your operator available in a catalog, or you've chosen an operator from an existing catalog, you can install your operator by creating a Subscription to a specific channel.
```
Comment (jmrodri) [inline content/en/docs/_index.md:82]: Again not necessarily a "building block of OLM". This is more of a step of what a person would do to get their operator running in a cluster.

**Review by jmrodri — CHANGES_REQUESTED** [review]
These changes make the page way too dense and confusing. We should either create a new page for this information or remove the existing contents of this index page in favor of these developers/admins solution. Though I think a new page would make more sense. 

The developer section has good information, but it is missing HOW OLM solves problems for the developer/admin which is the presumed audience.

Also, the building blocks section isn't much "building block" but more of a set of steps on what a user would do to get their operator installed from OLM.

File: content/en/docs/_index.md [hunk 4]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
 ### Declarative UI controls
 Operators can behave like managed service providers. Their user interface on the command line are APIs. For graphical consoles OLM annotates those APIs with descriptors that drive the creation of rich interfaces and forms for users to interact with the Operator in a natural, cloud-like way. 
 
+
+## How OLM solved the cluster admin and developer needs.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 5]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Operator Installation and Management
+
+As there are multiple steps involved in deploying an Operator, including creating the deployment, adding the custom resource definitions, and configuring the necessary permissions, a management layer becomes necessary to facilitate the process.
+
+Operator Lifecycle Manager (OLM) fulfills this role by introducing a packaging mechanism for delivering Operators and the necessary metadata for visualizing them in compatible UIs, including installation instructions and API hints in the form of CRD descriptors.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 6]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Dependency Resolution
+
+OLM manages the dependency resolution and upgrade lifecycle of running operators. 
+
+If a dependency is ever discovered, OLM will attempt to resolve said dependency by searching through known `CatalogSources` for an operator that `owns` the API. If an operator is found that `owns`the `required` API, OLM will attempt to deploy the operator that `owns` the API. If no `owner` is found, OLM will fail to deploy the operator with the dependency.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 7]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Work similar to package Managers
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 8]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+OLM verifies that all require APIs are available for the operator. If it is not available, then it will not install the operator.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 9]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+OLM will never update an operator in a way that breaks another that depends upon it.
+
+### Operator Groups
+
+OLM introduces the concept of `OperatorGroups` to enable cluster admins complete control over the permissions that OLM grants operators that it deploys.
+
+With `OperatorGroups`, a cluster admin can:
+
+* Define the set of permissions that OLM may grant to member operators
+* Define the set of namespaces that OLM may grant namespaced permissions in.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to above suggestion.

File: content/en/docs/_index.md [hunk 10]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+## Building blocks of OLM
+
+### [Install OLM](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/getting-started/_index.md)
+
+OLM requires you to provide metadata about your operator in order to ensure that it can be kept running safely on a cluster, and to provide information about how updates should be applied as you publish new versions of your operator. This is very similar to packaging software for a traditional operating system.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Agree, these are moreover steps for the operator installation.

File: content/en/docs/_index.md [hunk 11]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### [Package Validation](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/validate-package.md)
+
+Once you've [created your operator's package manifests](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/creating-operator-manifests.md), you will want to ensure that your package is valid and in the correct format. To ensure this, you should take several steps to ensure that your package can be used to install your operator in OLM.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md:62]: Agree, these are moreover steps for the operator installation.

File: content/en/docs/_index.md [hunk 12]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### [Publish Package into catalog](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/make-operator-part-of-catalog.md)
+
+We will publish the package into a catalog, install that catalog onto a Kube cluster, and then install the operator onto that cluster. If all of that succeeds and your operator is behaving as expected, your package is valid.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md:66]: Agree, these are moreover steps for the operator installation.

File: content/en/docs/_index.md (hunk 3 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Agree, these are moreover steps for the operator installation.

File: content/en/docs/_index.md [hunk 13]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### [Install Operator](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/install-operator-with-olm.md)
+
+Once you've made your operator available in a catalog, or you've chosen an operator from an existing catalog, you can install your operator by creating a Subscription to a specific channel. 
+
+### [Uninstall Operator](https://github.com/laxmikantbpandhare/olm-docs/blob/master/content/en/docs/Tasks/uninstall-operator.md)
+
+Delete subscriptions and ClusterServiceVersion. Both `Subscription` and `CSV` are namespaced objects meaning you need to delete a `Subscription` and a `CSV` in a specific namespace where you install the operator into. 
+
+By deleting `ClusterServiceVersion`, it will delete the operator's resources that OLM created for the operator such as deployment, pod(s), RBAC, and others.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Agree, these are moreover steps for the operator installation.

File: content/en/docs/_index.md (hunk 2 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Yup, I will add more concepts that exist in OLM and their main purpose.

File: content/en/docs/_index.md [hunk 14]
```diff
@@ -29,5 +29,60 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Works similar to package Managers
+
+Operator developers may need to use one or more APIs for the operator installation and OLM helps solve this use case by verifying that all of the required APIs are available for the operator. If they are not available, then OLM will not install the operator.
+
+### Operator Groups
+
+To give cluster admins complete control over the permissions that OLM grants operators, OLM introduced the concept of `OperatorGroups`.
+
+* Define the set of permissions that OLM may grant to member operators
```
Comment (marc-obrien) [inline content/en/docs/_index.md:51]: Defines the set of permissions that OLM may grant to member operators

File: content/en/docs/_index.md [hunk 15]
```diff
@@ -29,5 +29,60 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+* Define the set of namespaces that OLM may grant namespaced permissions in.
```
Comment (marc-obrien) [inline content/en/docs/_index.md:52]: Defines the set of namespaces that OLM may grant namespaced permissions in.

**Review by marc-obrien — APPROVED** [review]
/lgtm

File: content/en/docs/_index.md (hunk 15 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified as per suggestion

File: content/en/docs/_index.md (hunk 14 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified as per suggestion

File: content/en/docs/_index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: I discussed this suggestion with docs-pod and we decided to split the page. We will work on this at the end.

File: content/en/docs/_index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Added the appropriate building blocks.

File: content/en/docs/_index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Added the appropriate building blocks.

File: content/en/docs/_index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Added the appropriate building blocks.

File: content/en/docs/_index.md (hunk 3 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Added the appropriate building blocks.

File: content/en/docs/_index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Added the appropriate building blocks.

File: content/en/docs/_index.md [hunk 16]
```diff
@@ -29,5 +29,58 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
 ### Declarative UI controls
 Operators can behave like managed service providers. Their user interface on the command line are APIs. For graphical consoles OLM annotates those APIs with descriptors that drive the creation of rich interfaces and forms for users to interact with the Operator in a natural, cloud-like way. 
 
+
+## How OLM solves the cluster admin and developer needs.
+
+### Operator Installation and Management
+
+This packaging mechanism helps to simplify the multiple steps involved in deploying an Operator. OLM fulfills necessary metadata for visualizing them incompatible UIs, including installation instructions and API hints in the form of CRD descriptors. It helps cluster admin, developers to manage the lifecycle of all Operators and their associated services running across their clusters.
```
Comment (jmrodri) [inline content/en/docs/_index.md:37]:
```suggestion
This packaging mechanism helps to simplify the multiple steps involved in deploying an Operator. OLM fulfills necessary metadata for visualizing them in compatible UIs, including installation instructions and API hints in the form of CRD descriptors. It helps cluster admin, developers to manage the lifecycle of all Operators and their associated services running across their clusters.
```

**Comment by jmrodri** [comment]
@laxmikantbpandhare one typo incompatible -> in compatible. But looks 100% better, great work.

**Review by jmrodri — APPROVED** [review]

File: content/en/docs/_index.md (hunk 16 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: modified.

**Pull request merged**